yt/data_objects/construction_data_containers.py

```diff
@@ -1290,35 +1290,46 @@ class YTSmoothedCoveringGrid(YTCoveringGrid):
     _min_level = None
 
     @wraps(YTCoveringGrid.__init__)
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, *, kind="linear", nbuf=None, **kwargs):
         ds = kwargs["ds"]
         self._base_dx = (
             ds.domain_right_edge - ds.domain_left_edge
         ) / ds.domain_dimensions.astype("float64")
         self.global_endindex = None
         YTCoveringGrid.__init__(self, *args, **kwargs)
         self._final_start_index = self.global_startindex
+        # Assign order (covergence) based on interpolation kind
+        order = {
+            "natural": -2,
+            "akima": -1,
+            "linear": 2,
+            "cubic": 4,
+            "quintic": 6,
+        }[kind]
+        # Assess buffer region surrounding grid
+        if nbuf is None:
+            if order > 0:
+                nbuf = order // 2
+            else:
+                nbuf = 3
+        elif nbuf < 1:
+            raise ValueError
```

Comment:
I love this chunk!

I think this would fail with a `KeyError`, say if I pass `kind="dinosaur"`. Would it be possible to catch this and raise a more useful exception as this is directly user-facing?

```suggestion
       order_nbuf = {
            "natural": (-2, 3),
            "akima": (-1, 3),
            "linear": (2, 1),
            "cubic": (4, 2),
            "quintic": (6, 3)
        }
        try:
            order, nbuf_from_order = order_nbuf[kind]
        except KeyError:
            msg = f"Interpolation '{kind}' has not been implemented. "
            msg += "Only the following are supported: "
            msg += ", ".join(order_nbuf.keys())
            raise NotImplementedError(msg)
        nbuf = nbuf or nbuf_from_order
        if nbuf < 1:
            raise ValueError
```

Reply:
the raise statement should be a `raise from`. Other than that, great suggestion

Reply:
@cphyc I really like this as well! @neutrinoceros , which line exactly do you mean? Both raise statements?

Reply:
I mean the nested raise.
The difference is that `raise from` clearly indicates that the original error wasn't meant to be handled, but rather wrapped, while if you just raise an error directly in à `except` block, the error message looks like the new error wasn't intended.

Reply:
Got it. So we should change line 1311 to `except KeyError as k` and line 1315 to `raise NotImplementedError(msg) from k`, right?

Reply:
Or something like that.